packages/next/next-server/server/api-utils.ts

```diff
@@ -235,7 +235,9 @@ export function redirect(
     url = statusOrUrl
     statusOrUrl = 307
   }
-
+  if (typeof statusOrUrl === 'number' && typeof url !== 'string') {
```

Comment:
```suggestion
  if (typeof url !== 'string') {
```

Since this `if` happens after the first check, `statusOrUrl` is always a number at this point

Reply:
I agree with changes.